File: src/main/java/org/springframework/samples/petclinic/rest/mapper/FeedbackMapper.java
package org.springframework.samples.petclinic.rest.mapper;

import java.util.List;

import org.mapstruct.Mapper;
import org.springframework.samples.petclinic.feedback.Feedback;
import org.springframework.samples.petclinic.rest.dto.FeedbackCreateDto;
import org.springframework.samples.petclinic.rest.dto.FeedbackDto;

@Mapper(componentModel = "spring")
public interface FeedbackMapper {

	FeedbackDto toDto(Feedback feedback);

	List<FeedbackDto> toDtoList(List<Feedback> feedbackList);

	Feedback toEntity(FeedbackCreateDto dto);

}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/mapper/FeedbackMapper.kt
package org.springframework.samples.petclinic.rest.mapper

import org.springframework.samples.petclinic.feedback.Feedback
import org.springframework.samples.petclinic.rest.dto.FeedbackCreateDto
import org.springframework.samples.petclinic.rest.dto.FeedbackDto
import org.springframework.stereotype.Component

@Component
class FeedbackMapper {

    fun toDto(feedback: Feedback): FeedbackDto {
        return FeedbackDto(
            id = feedback.id,
            name = feedback.name,
            email = feedback.email,
            message = feedback.message,
            createdAt = feedback.createdAt
        )
    }

    fun toDtoList(feedbackList: List<Feedback>): List<FeedbackDto> {
        return feedbackList.map { toDto(it) }
    }

    fun toEntity(dto: FeedbackCreateDto): Feedback {
        val feedback = Feedback()
        feedback.name = dto.name
        feedback.email = dto.email
        feedback.message = dto.message
        return feedback
    }

}